applicationset/controllers/applicationset_controller.go

```diff
@@ -157,8 +157,13 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// appSyncMap tracks which apps will be synced during this reconciliation.
 	appSyncMap := map[string]bool{}
 
+	strategy, err := r.getApplicationSetStrategySpec(ctx, &applicationSetInfo)
+	if err != nil {
+		log.Errorf("failed to get application set syncstrategy: %v", err)
+	}
+
```

Comment:
Maybe set the strategy to nil initially and update it with this call inside the `if r.EnableProgressiveSyncs {` this way everything is behind the feature flag.

Reply:
Done